### PR TITLE
Reclaim Enhancement: Add Reclaimable parameter for queue

### PR DESCRIPTION
**Pull request by sivanzcw**
* Add parameter for queue to support queue unReclaimable

### CODE REVIEW

File: pkg/apis/scheduling/types.go [hunk 1]
```diff
@@ -261,6 +261,8 @@ type QueueSpec struct {
 	Capability v1.ResourceList
 	// State controller the status of queue
 	State QueueState
+	// UnReclaimAble indicate whether the queue can be reclaimed by other queue
+	UnReclaimAble bool
```
Comment (k82cn) [inline pkg/apis/scheduling/types.go]: let's use `Reclaimable` directly :)

File: pkg/scheduler/actions/reclaim/reclaim.go [hunk 2]
```diff
@@ -142,6 +142,10 @@ func (alloc *reclaimAction) Execute(ssn *framework.Session) {
 				if j, found := ssn.Jobs[task.Job]; !found {
 					continue
 				} else if j.Queue != job.Queue {
+					q := ssn.Queues[j.Queue]
+					if q != nil && q.Queue != nil && true == q.Queue.Spec.UnReclaimAble {
```
Comment (k82cn) [inline pkg/scheduler/actions/reclaim/reclaim.go]: that should be fine to check `Reclaimable` directly.

File: pkg/apis/scheduling/types.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/apis/scheduling/types.go]: I assume you want to make it compatible with old versions. 
make it a pointer type will be ok

File: pkg/apis/scheduling/v1alpha2/types.go [hunk 3]
```diff
@@ -262,6 +262,8 @@ type QueueSpec struct {
 	Capability v1.ResourceList `json:"capability,omitempty" protobuf:"bytes,2,opt,name=capability"`
 	// State controller the status of queue
 	State QueueState `json:"state,omitempty" protobuf:"bytes,3,opt,name=state"`
+	// ReclaimAble indicate whether the queue can be reclaimed by other queue
+	ReclaimAble *bool `json:"reclaimable,omitempty" protobuf:"bytes,4,opt,name=unreclaimable"`
```
Comment (hzxuzhonghu) [inline pkg/apis/scheduling/v1alpha2/types.go]: s/unreclaimable/reclaimable

File: pkg/apis/scheduling/types.go [hunk 4]
```diff
@@ -261,6 +261,8 @@ type QueueSpec struct {
 	Capability v1.ResourceList
 	// State controller the status of queue
 	State QueueState
+	// ReclaimAble indicate whether the queue can be reclaimed by other queue
+	ReclaimAble *bool
```
Comment (k82cn) [inline pkg/apis/scheduling/types.go]: s/ReclaimAble/Reclaimable/g

File: pkg/scheduler/actions/reclaim/reclaim.go [hunk 5]
```diff
@@ -142,6 +142,10 @@ func (alloc *reclaimAction) Execute(ssn *framework.Session) {
 				if j, found := ssn.Jobs[task.Job]; !found {
 					continue
 				} else if j.Queue != job.Queue {
+					q := ssn.Queues[j.Queue]
+					if false == q.ReclaimAble() {
```
Comment (k82cn) [inline pkg/scheduler/actions/reclaim/reclaim.go]:
```
if !q.Reclaimable() {
    ...
}
```

File: pkg/scheduler/api/queue_info.go [hunk 6]
```diff
@@ -56,3 +56,20 @@ func (q *QueueInfo) Clone() *QueueInfo {
 		Queue:  q.Queue,
 	}
 }
+
+// ReclaimAble return whether queue is reclaimAble
+func (q *QueueInfo) ReclaimAble() bool {
```
Comment (k82cn) [inline pkg/scheduler/api/queue_info.go]: It's better to set default value in admission controller; so user can also check this value in cli :)

Comment (sivanzcw) [reply pkg/scheduler/api/queue_info.go]: Since there is no queue admission now, this may require another commit to handle queue validating and mutating.

Comment (k82cn) [reply pkg/scheduler/api/queue_info.go]: yes, that's ok to open another PR for that :)

**Pull request merged**